packages/theme-check-common/src/checks/liquid-html-syntax-error/index.ts

```diff
@@ -61,3 +66,33 @@ export const LiquidHTMLSyntaxError: LiquidCheckDefinition = {
     };
   },
 };
+
+function detectInvalidBooleanExpressions(
+  context: Context<SourceCodeType.LiquidHtml>,
+): Check<SourceCodeType.LiquidHtml> {
+  return {
+    async BooleanExpression(node) {
+      const condition = node.condition;
+
+      if (
+        condition.type !== NodeTypes.Comparison &&
+        condition.type !== NodeTypes.LogicalExpression
+      ) {
+        return;
+      }
```

Comment:
It's excellent that we're identifying comparison and logical expressions, but since `{{ a == b }}` is just as invalid as `{{ a === b }}` today, I think we should lint both. Otherwise, we may lead users on thinking the second scenario is valid:

![scenario](https://github.com/user-attachments/assets/23cd8218-1773-406f-b4f3-497fbf652c9b)